### PR TITLE
[Serverless] Fix runtime detection

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

This PR fixes runtime detection.
Formerly we skipped pid = 1 to avoid detecting AWS_Lambda_rapid as runtime (since this is the internal API runtime name in AWS Lambda ecosystem). .NET integration tests have shown that this is not reliable so this PR fixes this behaviour. 

### Additional Notes

Screen Shot 2021-11-22 at 11 17 26 AM

### Describe how to test/QA your changes

Unit test + integration tests

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functi
onality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by IvanTopolcic — APPROVED** [review]
LGTM for after code freeze 👍

**Comment by nhinsch** [comment]
Before merging, let's figure out which branch we want to merge this to. We might want to make a release branch for the next release of the extension.

**Pull request merged**
